### PR TITLE
Log and instrument UserSocket connect

### LINKED ISSUE #3046: Log and instrument UserSocket.connect

### PR DESCRIPTION
**Pull request by chrismccord**
Closes #3046

### CODE REVIEW

File: lib/phoenix/socket.ex [hunk 1]
```diff
@@ -520,6 +546,10 @@ defmodule Phoenix.Socket do
     :ok
   end
 
+  @doc false
+  def __log__(Phoenix.LiveReloader.Socket, _opts), do: false
```
Comment (josevalim) [inline lib/phoenix/socket.ex]: We shouldn't hardcode this here. Instead we should set log: false in the live reloader lib and ship a new version.

Comment (chrismccord) [reply lib/phoenix/socket.ex]: Agree, which we will do, but folks who don't `mix deps.update phoenix_live_reload` will get spammy logs. It's not ideal to match on this here, but I think we do something similar elsewhere for ergonomics. Thoughts?

Comment (josevalim) [reply lib/phoenix/socket.ex]: Then they can update their `phoenix_live_reload`. :)

Comment (chrismccord) [reply lib/phoenix/socket.ex]: Fair :D

**Pull request merged**